dattri/algorithm/influence_function.py

```diff
@@ -620,35 +649,94 @@ def transform_test_rep(
+            grad: torch.Tensor,
+            regularization: float,
+        ) -> torch.Tensor:
+            """Transformation of a single test representation.
+
+            For any test representation v and training gradient grad, along
+            with regularization term r, DataInf gives:
+            q = (v - (dot(v,grad) / (r + torch.norm(grad) ** 2)) * grad) / r
+
+            Transformed test representations are later used for influence calculation:
+            Inf = dot(g,q) where g is a training representation and q is a transformed
+            test representation.
+
+            Args:
+                v (torch.Tensor): A tensor representing (batched) test set
+                    representation, of shape (num_test,parameter_size).
+                grad (torch.Tensor): A tensor representing a single training
+                    gradient, of shape (parameter_size,).
+                regularization (float): A float default to 0.1. Specifies
+                    the regularization term to be added to the empirical Fisher
+                    information matrix in each layer.
+
+            Returns:
+                torch.Tensor: A tensor corresponding to transformed test representation.
+            """
+            grad = grad.unsqueeze(-1)
+            coef = (v @ grad) / (regularization + torch.norm(grad) ** 2)
+            return (v - coef @ grad.T) / regularization
+
+        regularization = self.transformation_kwargs["regularization"]
+        # Split layer-wise train and test representations
+        query_split = self.get_layer_wise_reps(ckpt_idx, test_rep)
+        full_rep_split = self.get_layer_wise_reps(ckpt_idx,
+         self._cached_train_reps[f"ckpt_{ckpt_idx}"])
+        layer_cnt = len(full_rep_split)
+        ifvps = []
+        # Use test batch size as intermediate batch size
+        # Peak memory usage: max(train_batch_size,test_batch_size)*test_batch_size*p
+        batch_size = test_rep.shape[0]
+        for layer in range(layer_cnt):
+            grad_layer = full_rep_split[layer]
+            # length = full train data size
+            length = grad_layer.shape[0]
+            # Split gradients into smaller chunks
+            grad_batches = grad_layer.split(batch_size, dim=0)
+            running_contributions = 0
+            for batch in grad_batches:
+                reg = 0.1 if regularization is None else regularization
+                contribution = torch.func.vmap(
+                    lambda grad, layer=layer, reg=reg: _transform_single_test_rep(
+                    query_split[layer],
+                    grad,
+                    reg,
+                ),
+                )(batch)
+                # Sum up the contributions by chunk and average at the end
+                running_contributions += contribution.sum(dim=0)
+            running_contributions /= length
+            ifvps.append(running_contributions)
+        return torch.cat(ifvps, dim=1)
+
+    def get_layer_wise_reps(self,
```

Comment:
How about name this as _get_layer_wise_reps since this is a minor helper function